src/DetermineRoot.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace DigipolisGent\Robo\Task\General;
+
+use Robo\Result;
+use Robo\Task\BaseTask;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Determines the root of a Drupal project and adds it to the config.
+ * Config is available for classes implementing
+ * \Robo\Contract\ConfigAwareInterface and using \Robo\Common\ConfigAwareTrait.
+ *
+ * ``` php
+ * $this->taskDetermineRoot()
+ *     ->dir(getcwd())
+ *     ->run();
+ * $this->getConfig()->get('digipolis.root.project');
+ * ```
+ *
+ */
+class DetermineRoot extends BaseTask
+{
+    /**
+     * The directory in which to search for the project root.
+     *
+     * @var string
+     */
+    protected $dir;
+
+    /**
+     * The maximum depth to traverse directories.
+     *
+     * @var int
+     */
+    protected $depth;
+
+    /**
+     * The Drupal finder to use to find the project root.
```

Comment:
In theory we can, but that would mean we'd have to create a class that extends the symfony finder and implements the interface (without actually overwriting any of the methods)... Otherwise we can't typehint on the interface we created. The symfony finder itself does not implement any 'finder' interface.